module/VuFind/src/VuFind/I18n/Sorter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Class Sorter
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Moravian Library 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  I18n
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\I18n;
+
+/**
+ * Class Sorter
+ *
+ * @category VuFind
+ * @package  I18n
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Sorter implements SorterInterface
+{
+    /**
+     * Intl Collator
+     *
+     * @var \Collator
+     */
+    protected $collator;
+
+    /**
+     * Do respect current locale?
+     *
+     * @var bool
+     */
+    protected $respectLocale;
+
+    /**
+     * Constructor
+     *
+     * @param string $locale        Current user locale
+     * @param bool   $respectLocale Do respect current locale?
+     */
+    public function __construct(string $locale = 'en', bool $respectLocale = false)
+    {
+        $this->collator = new \Collator($locale);
```

Comment:
I think Sorter should take Collator as a constructor parameter instead of instantiating it. This would allow setting further options for the Collator in the factory instead of overriding Sorter class.

Reply:
I agree, this would be more flexible. If you want to have an easy way to build a Sorter without having to also construct a Collator, you could create a static method like `buildSorterForLocale` that could be used as an alternate constructor (e.g. for more concise Sorter construction in unit tests).

Reply:
I've changed it as you suggest